File: data-import/channels/channels.js
import Medusa from "@medusajs/medusa-js"

const MEDUSA_BACKEND_URL = "http://localhost:7001"

const medusa = new Medusa({
  baseUrl: MEDUSA_BACKEND_URL,
})

// must be previously logged in or use api token
medusa.admin.users.update(userId, {
  api_token,
})
.then(({ user }) => {
  console.log(user.api_token)
})

// console.log("STARTED")

//TODO
//Determine how to create sales channels
//should start with two by store, desktop and mobile

//TODO
// //Use fetch to create sales channel
// fetch(`{${BACKEND_URL}/admin/sales-channels`, {
//     method: "POST",
//     credentials: "include",
//     headers: {
//       "Content-Type": "application/json",
//     },
//     body: JSON.stringify({
//       name: "App",
//       description: "Mobile app",
//     }),
//   })
//   .then((response) => response.json())
//   .then(({ sales_channel }) => {
//     console.log(sales_channel.id)
//   })

//TODO
//add products to sales channel

// fetch(
//     `<BACKEND_URL>/admin/sales-channels/${salesChannelId}/products/batch`,
//     {
//       method: "POST",
//       credentials: "include",
//       headers: {
//         "Content-Type": "application/json",
//       },
//       body: JSON.stringify({
//         product_ids: [
//           {
//             id: productId,
//           },
//         ],
//       }),
//     }
//   )
//   .then((response) => response.json())
//   .then(({ sales_channel }) => {
//     console.log(sales_channel.id)
//   })

File: data-import/channels/channels.py
import requests
import json

BACKEND_URL = "http://localhost:9000"
EMAIL = "drexel-sd-shopping@gmail.com"
PASSWORD = "SeniorDesign#1234"

s = requests.Session();

headers = {'Content-Type': 'application/json'}

## Authorize session
url = f"{BACKEND_URL}/admin/auth"
payload = {"email": EMAIL, "password": PASSWORD}
res = s.post(url, headers=headers, json=payload)

SALES_CHANNEL_NAMES = [
  "Test Channel 1",
  "Test Channel 2",
  "Test Channel 3"
]

## create sales channels
sales_channel_ids = {}
for name in SALES_CHANNEL_NAMES:
  url = f"{BACKEND_URL}/admin/sales-channels"
  payload = {"name": f"{name}", "description": f"{name} description"}
  res = s.post(url, headers=headers, json=payload)
  res_json = res.json()["sales_channel"];
  sales_channel_ids[name] = res_json['id']
  print(f"Created sales channel {name}: {res_json['id']}")


API_KEY_NAME = "Test API Key"
## create publishable api key
url = f"{BACKEND_URL}/admin/publishable-api-keys"
payload = {"title": f"{API_KEY_NAME}"}
res = s.post(url, headers=headers, json=payload)
pak_json = res.json()["publishable_api_key"]
print("API KEY CREATED")

## link sales channels to api key
url = f"{BACKEND_URL}/admin/publishable-api-keys/{pak_json["id"]}/sales-channels/batch"
parts = []
for i, sales_channel in enumerate(sales_channel_ids.items()):
  parts.append({"id": f"{sales_channel[1]}"})

payload = {"sales_channel_ids": parts}
res = s.post(url, headers=headers, json=payload)
print(f"SALES CHANNEL LINK RES: {res}")